### PR TITLE
Add tls client options to api/cli

### PR DESCRIPTION
**Pull request by kyhavlov**
This adds TLS client options to the api and cli for providing client certs and setting the TLS server name when connecting to Consul.

### CODE REVIEW

File: command/base/command.go [hunk 1]
```diff
@@ -83,6 +94,19 @@ func (c *Command) httpFlagsClient(f *flag.FlagSet) *flag.FlagSet {
 		f = flag.NewFlagSet("", flag.ContinueOnError)
 	}
 
+	f.Var(&c.caFile, "ca-file",
+		"Path to a CA file to use for TLS when communicating with Consul. This "+
+			"can also be specified via the CONSUL_CACERT environment variable.")
+	f.Var(&c.caPath, "ca-path",
+		"Path to a directory of CA certificates to use for TLS when communicating "+
+			"with Consul. is enabled. This can also be specified via the CONSUL_CAPATH "+
```
Comment (slackpad) [inline command/base/command.go]: "when \`verify_incoming\` is enabled"

Comment (kyhavlov) [inline command/base/command.go]: shouldn't the "is enabled." just be removed here? the CA settings can be used even when `verify_incoming` isn't enabled

Comment (slackpad) [inline command/base/command.go]: that's true

File: api/api.go [hunk 2]
```diff
@@ -347,6 +368,41 @@ func NewClient(config *Config) (*Client, error) {
 		config.HttpClient = defConfig.HttpClient
 	}
```
Comment (arvenil) [inline api/api.go]: So, with below I can pass my own `HttpClient`, with my own `RoundTripper`

```go
if config.HttpClient == nil {
    config.HttpClient = defConfig.HttpClient
}
```

but the line
```go
config.HttpClient.Transport.(*http.Transport).TLSClientConfig = tlsClientConfig
```

brakes it as it assumes standard `RoundTripper` implementation

**Comment by arvenil** [comment]
One way to solve that is to add new func `NewHttpClient` that will return ready to use consul http client. Then user can use/modify it and pass to `NewClient()` via config. Now `NewClient` would either use provided HttpClient or if nil then run `NewHttpClient()` to create default one. This means I could always provide 1) my own client 2) my own client based on default consul client 3) default consul client; and consul will never ever mess with client provided by user.

**Pull request merged**